src/components/tab-bar/tab-bar.tsx

```diff
@@ -0,0 +1,142 @@
+import {
+    Component,
+    h,
+    Prop,
+    Element,
+    State,
+    EventEmitter,
+    Event,
+} from '@stencil/core';
+import { MDCTabBar, MDCTabBarActivatedEvent } from '@limetech/mdc-tab-bar';
+import { strings } from '@limetech/mdc-tab-bar/constants';
+import { Tab } from './tab.types';
+const { TAB_ACTIVATED_EVENT } = strings;
+
+@Component({
+    tag: 'limel-tab-bar',
+    styleUrl: 'tab-bar.scss',
+    shadow: true,
+})
+export class TabBar {
+    @Element()
+    private element: HTMLElement;
+
+    /**
+     * List of tabs to display
+     */
+    @Prop()
+    public tabs: Tab[];
+
+    private mdcTabBar: MDCTabBar;
+
+    @Event()
+    private changeTab: EventEmitter<Tab>;
+
+    @State()
+    private selectedTab = this.tabs[0];
+
+    constructor() {
+        this.handleTabActivated = this.handleTabActivated.bind(this);
+        this.renderTab = this.renderTab.bind(this);
+    }
+
+    public componentDidLoad() {
+        this.setup();
+    }
+
+    private setup() {
+        const element = this.element.shadowRoot.querySelector('.mdc-tab-bar');
+        if (!element) {
+            return;
+        }
+
+        this.mdcTabBar = new MDCTabBar(element);
+
+        // Workaround for shadow dom support for material
+        (this
+            .mdcTabBar as any).foundation_.adapter_.getFocusedTabIndex = () => {
+            const tabElements = this.getTabElements();
+            const activeElement = this.element.shadowRoot.activeElement;
+            return tabElements.indexOf(activeElement);
+        };
+
+        this.setupListeners();
+    }
+
+    private getTabElements() {
+        return [].slice.call(
+            this.element.shadowRoot.querySelectorAll('.mdc-tab')
+        );
+    }
+
+    private setupListeners() {
+        this.mdcTabBar.listen(TAB_ACTIVATED_EVENT, this.handleTabActivated);
+    }
+
+    private handleTabActivated(event: MDCTabBarActivatedEvent) {
+        const index = event.detail.index;
+        const selectedTab = this.tabs[index];
+        this.selectedTab = selectedTab;
+        this.changeTab.emit({ ...this.selectedTab, active: true });
```

Comment:
The list component emits two events when selection changes, one for the old item telling the consumer that the item is no longer selected and one for the new selected item. We should probably do the same here :thinking: